flytekit/taskplugins/notebook/task.py

```diff
@@ -0,0 +1,215 @@
+
+T = typing.TypeVar("T")
+
+
+def _dummy_task_func():
+    return None
+
+
+class NotebookTask(PythonInstanceTask[T]):
+    """
+    Simple Papermill based input output handling for a Python Jupyter notebook. This task should be used to wrap
+    a Notebook that has 2 properties
+    Property 1:
+    One of the cells (usually the first) should be marked as the parameters cell. This task will inject inputs after this
+    cell. The task will inject the outputs observed from Flyte
+
+    Property 2:
+    For a notebook that produces outputs, that should be consumed by a subsequent notebook, use the method
+    :py:func:`record_outputs` in your notebook after the outputs are ready and pass all outputs.
+
+    Usage:
+
+    .. code-block:: pthon
+
+        val_x = 10
+        val_y = "hello"
+
+        ...
+        # cell begin
+        from flytekit.taskplugins.notebook import record_outputs
+
+        record_outputs(x=val_x, y=val_y)
+        #cell end
+
+    Step 2: Wrap in a task
+    Now point to the notebook and create an instance of :py:class:`NotebookTask` as follows
+
+    Usage:
+
+    .. code-block:: python
+
+        nb = NotebookTask(
+            name="modulename.my_notebook_task", # the name should be unique within all your tasks, usually it is a good
+                                               # idea to use the modulename
+            notebook_path="../path/to/my_notebook",
+            inputs=kwtypes(v=int),
+            outputs=kwtypes(x=int, y=str),
+            metadata=TaskMetadata(retries=3, cache=True, cache_version="1.0"),
+        )
+
+    Step 3: Task can be executed as usual
+
+    Outputs
+    -------
+    The Task produces 2 implicit outputs.
+
+    #. It captures the executed notebook in its entirety and is available from Flyte with the name ``out_nb``.
+    #. It also converts the captured notebook into an ``html`` page, which the FlyteConsole will render called -
+       ``out_rendered_nb``
+
+    .. note:
+
+        Users can access these notebooks after execution of the task locally or from remote servers.
+
+    .. todo:
+
+        Implicit extraction of SparkConfiguration from the notebook is not supported.
+
+    """
+
+    _IMPLICIT_OP_NOTEBOOK = "out_nb"
+    _IMPLICIT_OP_NOTEBOOK_TYPE = PythonNotebook
+    _IMPLICIT_RENDERED_NOTEBOOK = "out_rendered_nb"
+    _IMPLICIT_RENDERED_NOTEBOOK_TYPE = HTMLPage
+
+    def __init__(
+        self,
+        name: str,
+        notebook_path: str,
+        task_config: T = None,
+        inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        outputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        **kwargs,
+    ):
+        plugin_class = TaskPlugins.find_pythontask_plugin(type(task_config))
+        self._plugin = plugin_class(task_config=task_config, task_function=_dummy_task_func)
+        task_type = f"nb-{self._plugin.task_type}"
+        self._notebook_path = os.path.abspath(notebook_path)
```

Comment:
oh interesting, does this assume notebook tasks are only initialized in container?

Reply:
ya for the moment - they need the container